### PR TITLE
 Update middleware to use prefixDefaultLocale 

### PR DESCRIPTION
**Pull request by vannut**
With regards of [this issue](https://github.com/rainlab/translate-plugin/pull/318#issuecomment-354567683) this PR will add a check for the prefixDefaultLocale setting into the middleware.

As a result the said wrong behaviour will disappear :)

Thanks @msimkunas for speeding up the debugging a lot 👍

### CODE REVIEW

File: updates/version.yaml [hunk 1]
```diff
@@ -44,4 +44,5 @@
 1.3.2: Improved compatibility with RainLab.Pages, added ability to scan Mail Messages for translatable variables.
 1.3.3: Fix to the locale picker session handling in Build 420 onwards.
 1.3.4: Add alternate hreflang elements and adds prefixDefaultLocale setting.
-1.3.5: Fix MLRepeater bug when switching locales.
\ No newline at end of file
+1.3.5: Fix MLRepeater bug when switching locales.
+1.3.6: Fix Middleware to use the prefixDefaultLocale setting introduced in 134
```
Comment (LukeTowers) [inline updates/version.yaml]: Change to `introduced in 1.3.4`

File: classes/LocaleMiddleware.php [hunk 2]
```diff
@@ -12,15 +12,19 @@ class LocaleMiddleware
...
+        if( ! $translator->loadLocaleFromRequest()) {
+            if(Config::get('translate.prefixDefaultLocale')) {
+                $translator->loadLocaleFromSession();
+            } else {
+                $translator->setLocale($translator->getDefaultLocale());
+            }
+         }
 
-        return $next($request);
+         return $next($request);
```
Comment (LukeTowers) [inline classes/LocaleMiddleware.php]: remove extra whitespace

File: classes/LocaleMiddleware.php [hunk 3]
```diff
@@ -12,15 +12,19 @@ class LocaleMiddleware
...
+    public function handle($request, Closure $next)                                                                                                                                  
+    {                                                                                                                                                                                
         $translator = Translator::instance();
         $translator->isConfigured();
 
-        if (!$translator->loadLocaleFromRequest()) {
-            $translator->loadLocaleFromSession();
-        }
+        if( ! $translator->loadLocaleFromRequest()) {
+            if(Config::get('translate.prefixDefaultLocale')) {
```
Comment (LukeTowers) [inline classes/LocaleMiddleware.php]: Please use a space after if and before condition: i.e. `if (Config`. Also, `Config` has not been loaded into this scope so please import it with a `use Config;` statement at the top of the file. This code won't work without that change. Please ensure you test this thoroughly.

File: classes/LocaleMiddleware.php [hunk 4]
```diff
@@ -12,15 +12,19 @@ class LocaleMiddleware
      * @param  \Closure  $next
      * @return mixed
      */
-    public function handle($request, Closure $next)
-    {
+    public function handle($request, Closure $next)                                                                                                                                  
+    {
```
Comment (LukeTowers) [inline classes/LocaleMiddleware.php]: Please remove extra whitespace at end of above and this line.

**Comment by vannut** [comment]
Oh boy what a mess. I used the online github editor; I now can really appreciate the red waves in a proper IDE :)

**Comment by vannut** [comment]
Maybe @datune could check this update?

**Comment by datune** [comment]
@vannut Looks good to me, thanks for your work!

**Comment by osmanzeki** [comment]
@vannut @LukeTowers 
Quick question regarding this change. I'm getting a `null` value for the `prefixDefaultLocale` config in the `LocaleMiddleware` and it is causing the `setLocale` calls not to be saved. Am I missing something here?

This line specifically:
https://github.com/rainlab/translate-plugin/pull/326/files#diff-ed82a364afe7c7cbf84e4e750533bf82R22

screen shot 2018-01-11 at 4 04 12 pm

**Comment by osmanzeki** [comment]
When I get the value using this line:
```Config::get('rainlab.translate::prefixDefaultLocale')```

Everything seems to go work normally.

**Comment by LukeTowers** [comment]
@osmanzeki already identified and fixed in https://github.com/rainlab/translate-plugin/pull/329

**Comment by osmanzeki** [comment]
Ok thank you, I'll update my plugin locally.

**Pull request merged**